adapters/moscow.js

```diff
@@ -0,0 +1,238 @@
+  return function (done) {
+    request({
+      uri: link,
+      encoding: 'binary'
+    }, (err, res, data) => {
+      if (err || res.statusCode !== 200) {
+        return done(null, []);
+      }
+
+      data = new Buffer(data, 'binary');
+      var iconv = new Iconv('windows-1251', 'utf-8');
+      data = iconv.convert(data).toString();
+
+      let stationRegexp = /air-today\/station\/(\w*)\//;
+      let stationId = stationRegexp.exec(link)[1];
+
+      formatData(data, stationId, (measurements) => {
+        return done(null, measurements);
+      });
+    });
+  };
+};
+
+const formatData = function (data, stationId, cb) {
+  const $ = cheerio.load(data);
+
+  let regex = /«(.*)»/;
+  let match = regex.exec($('caption').text());
+  let location;
+  if (match) {
+    location = match[1].trim();
+  } else {
+    // this shouldn't happen
+    return cb([]);
+  }
+  // basically matches "М1 (Жубелино)"
+  let mobileRegexp = /[\u041c]{1}[\d]?[\s]?\([\u0400-\u04ff]*\)/;
+  // skip mobile stations for the time being
+  if (mobileRegexp.test(location)) {
+    return cb([]);
+  }
+
+  let base = {
+    city: 'Moscow',
+    location: location,
+    averagingPeriod: {unit: 'hours', value: 1},
+    attribution: [{
+      name: 'Mosecomonitoring',
+      url: 'http://mosecom.ru/air/'
+    }]
+  };
+
+  base = Object.assign(base, moscowLocations[stationId]);
+
+  let measurements = [];
+  let parameters = {};
+  let cols = 1;
+
+  $('th.header').each(function (i, e) {
+    // skip the date column
+    if (i === 0) {
+      return;
+    }
+    let param = $(this).text().split(' ')[0].toLowerCase();
+    if (acceptableParameters.indexOf(param) > -1) {
+      parameters[cols] = {
+        name: param,
+        unit: $('td.header').eq(cols - 1).text()
+      };
+    }
+    if ($(this).attr('colspan')) {
+      cols += Number($(this).attr('colspan'));
+    } else {
+      cols++;
+    }
+  });
+
+  // keep track of the previous full date
+  let lastFullDate;
+
+  $('tr.evenarg, tr.oddarg').each(function (i, e) {
+    let columns = $(this).children('td');
+
+    let dateText = $(columns).first().text();
+    let dateTime = getFullDate(dateText);
+    if (dateTime.isValid()) {
+      lastFullDate = dateTime;
+    } else {
+      if (lastFullDate) {
+        let dayTime = moment(dateText, 'HH:mm');
+        dateTime = moment.tz(
+          {
+            year: lastFullDate.year(),
+            month: lastFullDate.month(),
+            day: lastFullDate.day(),
+            hour: dayTime.hour(),
+            minute: dayTime.minute()
+          },
+          'Europe/Moscow');
+      } else {
+        // no known full date?
+        // you've got a problem
+      }
+    }
+
+    for (let i in parameters) {
+      if (parameters.hasOwnProperty(i)) {
+        let m = Object.assign({}, base);
+        m.parameter = parameters[i].name;
+        m.unit = parameters[i].unit;
+        if (m.unit === 'мг / куб. м') {
+          m.unit = 'mg/m³';
```

Comment:
Are we sure it's milli and not micro? @RocketD0g is confirming as well, but just want to make sure we get that right.